src/vaults/AsyncRequests.sol

```diff
@@ -142,8 +137,11 @@ contract AsyncRequests is BaseInvestmentManager, IAsyncRequests {
         uint128 shares_ = shares.toUint128();
         require(shares_ != 0, ZeroAmountNotAllowed());
 
-        // You cannot redeem using a disallowed asset, instead another vault will have to be used
-        require(poolManager.isLinked(vault_.poolId(), vault_.scId(), vault_.asset(), vault_), AssetNotAllowed());
+        VaultDetails memory vaultDetails = poolManager.vaultDetails(vault_);
+        PoolId poolId = vault_.poolId();
+        ShareClassId scId = vault_.scId();
+
+        require(poolManager.isLinked(poolId, scId, vaultDetails.asset, vault_), AssetNotAllowed());
```

Comment:
Nice one!

Reply:
I am wondering whether we should add the same check to the `SyncRequests.deposit`?